python/examples/atomistic/3-atomistic-model-with-nl.py

```diff
@@ -0,0 +1,433 @@
+# lattice constant of a = 3.641 Å. After initialization of the velocities, the system
+# was run for 100 ps with the same parameters we will use below and the final state can
+# be visualized as
+
+ax = ase.visualize.plot.plot_atoms(atoms, radii=0.5)
+ax.set_xlabel("$\\AA$")
+ax.set_ylabel("$\\AA$")
+plt.show()
+
+# %%
+#
+# The system already has velocities and the expected density of 1.374 g/cm³.
+
+u_to_g = 1.66053906660e-24
+Å_to_cm = 1e-08
+
+mass_density = sum(atoms.get_masses()) / atoms.cell.volume * u_to_g / Å_to_cm**3
+
+print(f"ρ_m = {mass_density:.3f} g/cm³")
+
+# %%
+#
+# Metatensor Neighbor Lists
+# -------------------------
+#
+# Before implementing the actual model, let's take a look at how metatensor stores
+# neighbor lists inside a :py:class:`System` object. We start by computing the neighbor
+# list for our argon systen using ASE.
+
+i, j, S, D = ase.neighborlist.neighbor_list(quantities="ijSD", a=atoms, cutoff=5.0)
+
+# %%
+#
+# The :py:func:`neighbor_list <ase.neighborlist.neighbor_list>` function returns the
+# neighbor indices: quantities ``"i"`` and ``"j"``, the neighbor shifts: ``"S"``, and
+# the distance vectors: ``"D"``. We now stack these together and convert them into the
+# suitable types.
+
+i = torch.from_numpy(i.astype(int))
+j = torch.from_numpy(j.astype(int))
+neighbor_indices = torch.vstack([i, j])
+neighbor_shifts = torch.from_numpy(S.astype(int))
+
+print("neighbor_indices:", neighbor_indices)
+print("neighbor_shifts:", neighbor_shifts)
+
+# %%
+# Create a Neighbor List
+# ^^^^^^^^^^^^^^^^^^^^^^
+#
+# We now assemble the neighbor list. First, we create the ``samples`` metadata for the
+# :py:class:`TensorBlock <metatensor.torch.TensorBlock>` which will hold the neighbor
+# list.
+
+sample_values = torch.hstack([neighbor_indices.T, neighbor_shifts])
+samples = Labels(
+    names=[
+        "first_atom",
+        "second_atom",
+        "cell_shift_a",
+        "cell_shift_b",
+        "cell_shift_c",
+    ],
+    values=sample_values,
+)
+
+neighbors = TensorBlock(
+    values=torch.from_numpy(D).reshape(-1, 3, 1),
+    samples=samples,
+    components=[Labels.range("xyz", 3)],
+    properties=Labels.range("distance", 1),
+)
+
+print(neighbors)
+
+# %%
+#
+# The data and metadata inside the ``neighbors`` object do not contain information about
+# the ``cutoff`` and whether this is a full or half neighbor list. To account for this,
+# metatensor neighbor lists are always stored together with
+# :py:class:`NeighborListOptions`. For our system, these options can be saved as
+
+options = NeighborListOptions(cutoff=5.0, full_list=True, requestor="ljmodel")
+
+# %%
+#
+# We set ``full_list=True`` because ASE computes a “full” list (as opposed to a half
+# list) where each `i, j` pair appears twice, stored once with `i` and once with `j`.
+# The ``requestor`` parameter is optional and can be used to further distinguish
+# neighbor lists if several are present.
```

Comment:
No, this is just recording who requested a given NL, but two NeighborListOptions with identical cutoff/full_list but different requestor will compare as equal. I would not mention this here